### PR TITLE
Added configuration controller to v2 API

### PR DESCRIPTION
**Pull request by rshbhgrg**
refs https://github.com/TryGhost/Ghost/issues/10060

### CODE REVIEW

File: core/server/api/v2/configuration.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+const Promise = require('bluebird'),
```
Comment (gargol) [inline core/server/api/v2/configuration.js]: this should be expanded to `const` per each line ;) as in [other](https://github.com/TryGhost/Ghost/blob/master/core/server/api/v2/settings.js#L1-L10) controller

File: core/test/functional/api/v2/admin/configuration_spec.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
+var should = require('should'),
+    supertest = require('supertest'),
+    testUtils = require('../../../../utils'),
+    localUtils = require('./utils'),
+    config = require('../../../../../../core/server/config'),
+    ghost = testUtils.startGhost,
+    request;
+
+describe('Configuration API', function () {
+    var accesstoken = '', ghostServer;
```
Comment (gargol) [inline core/test/functional/api/v2/admin/configuration_spec.js]: These two variables can be removed as they are never used anymore. Not sure this is a part of migration but I'm doing this cleanup for the tests that are being copied :)

**Comment by gargol** [comment]
Other than some non-crucial improvements it's good to :ship:

**Comment by gargol** [comment]
@rishabhgrg could u please update this one? would love to clean up our PR list :)

**Comment by rishabhgrg** [comment]
@kirrg001 / @gargol This is now rebased and conflicts fixed, if you have some time to look at this one today to further shrink #10060 😄

File: core/server/api/v2/configuration.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        environment: config.get('env'),
+        database: config.get('database').client,
+        mail: isPlainObject(config.get('mail')) ? config.get('mail').transport : ''
+    };
+}
+
+function getBaseConfig() {
+    return {
+        useGravatar: !config.isPrivacyDisabled('useGravatar'),
+        publicAPI: labs.isSet('publicAPI'),
```
Comment (kirrg001) [inline core/server/api/v2/configuration.js]: The admin client still uses the labs checkbox for the `publicAPI`. So we still need to serve this config 👍

File: core/server/api/v2/utils/serializers/output/configuration.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const debug = require('ghost-ignition').debug('api:v2:utils:serializers:output:mail');
```
Comment (kirrg001) [inline core/server/api/v2/utils/serializers/output/configuration.js]: > api:v2:utils:serializers:output:mail

🤩

**Review by kirrg001 — APPROVED** [review]
Feel free to self merge 👍

**Pull request merged**